typescript/raven.d.ts

```diff
@@ -1,287 +1,301 @@
+     *
+     * @param user The definition of the currently active user's unique identity
+     */
+    setUserContext(user: RavenUserContext): RavenStatic;
 
     /**
      * Clear the user context, removing the user data that would be sent to Sentry.
      */
     setUserContext(): RavenStatic;
 
-    /*
-     * Set a user to be sent along with the payload.
-     *
-     * @param {object} user An object representing user data [optional]
-     * @return {Raven}
+    /** 
+     * Add arbitrary data to be sent along with the payload.
+     * @param extra data of an arbitrary, nested type which will be added
      */
-    setUserContext(user: RavenUserContext): RavenStatic;
-
-    /** Merge extra attributes to be sent along with the payload. */
-    setExtraContext(context: Object): RavenStatic;
+    setExtraContext(extra: { [prop: string]: any }): RavenStatic;
 
-    /** Merge tags to be sent along with the payload. */
-    setTagsContext(tags: Object): RavenStatic;
+    /** 
+     * Add additional tags to be sent along with payloads.
+     * @param tags A key/value-pair which will be added
+     */
+    setTagsContext(tags: { [id: string]: string }): RavenStatic;
 
-    /** Clear all of the context. */
+    /** 
+     * Clear the whole currently set context.
+     */
     clearContext(): RavenStatic;
 
-    /** Get a copy of the current context. This cannot be mutated.*/
+    /** 
+     * Get a copy of the current context.
+     */
     getContext(): Object;
 
-    /** Override the default HTTP data transport handler. */
-    setTransport(transportFunction: (options: RavenTransportOptions) => void): RavenStatic;
-
-    /** Set environment of application */
+    /** 
+     * Set environment of application
+     * @param environment Typically something like 'production'
+     */
     setEnvironment(environment: string): RavenStatic;
 
-    /** Set release version of application */
+    /**
+     * Set release version of application
+     * @param release Typically something like a git SHA to identify the current version
+     */
     setRelease(release: string): RavenStatic;
 
-    /** Get the latest raw exception that was captured by Raven.*/
+    /**
+     * Specify a function that can mutate the payload right before it is being sent to Sentry.
+     * @param callback The function which can mutate the data
+     */
+    setDataCallback(callback: (data: any, orig?: string) => any): RavenStatic;
+
+    /** 
+     * Specify a callback function that can mutate or filter breadcrumbs when they are captured.
+     * @param callback The function which applies the filter
+     */
+    setBreadcrumbCallback(callback :(data: any, orig?: string) => any): RavenStatic;
+
+    /**
+     * Specify a callback function that determines if the given message should be sent to Sentry.
+     * @param callback The function which determines if the given blob should be sent
+     */
+    setShouldSendCallback(callback: (data: any, orig?: string) => boolean): RavenStatic;
```

Comment:
It sure is...
I will look into it again if it is (reasonably) possible to be more specific about the data passed to these various callbacks